modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/QuickLinksPanel/index.js

```diff
@@ -0,0 +1,134 @@
+import ClayIcon from '@clayui/icon';
+import DOMPurify from 'dompurify';
+import {useCallback, useEffect, useState} from 'react';
+import {fetchHeadless} from '../../../../common/services/liferay/api';
+import {useCustomerPortal} from '../../context';
+
+const QuickLinksPanel = () => {
+	const [{quickLinks}] = useCustomerPortal();
+	const [expandedPanel, setExpandedPanel] = useState(true);
+	const [quickLinksData, setQuickLinksData] = useState([]);
+
+	const fetchQuickLinksPanelContent = useCallback(async () => {
+		const webContentFolderName = 'actions';
+		const webContentTemplateName = 'Action Card';
+
+		const siteGroupId = Liferay.ThemeDisplay.getSiteGroupId();
+
+		const structuredContentFolders = await fetchHeadless({
+			url: `/sites/${siteGroupId}/structured-content-folders`,
+		});
+
+		const {id: quickLinksInstructionsFolderID} =
+			structuredContentFolders.items.find(
+				({name}) => name === webContentFolderName
+			) || {};
+
+		const contentTemplates = await fetchHeadless({
+			url: `/sites/${siteGroupId}/content-templates`,
+		});
+
+		const contentTemplate = contentTemplates.items.find(
+			({name}) => name === webContentTemplateName
+		);
+
+		const structuredContents = await fetchHeadless({
+			url: `/structured-content-folders/${quickLinksInstructionsFolderID}/structured-contents`,
+		});
+
+		const renderedQuickLinksData = await quickLinks.reduce(
```

Comment:
why this reduce is using await? it's possible??

From what I know, async/await doesn't work inside of array functions, like: reduce, map, forEach etc